reporters/reporter-default/src/main/java/org/radargun/reporting/html/IndexDocument.java

```diff
@@ -218,71 +110,58 @@ private void addToConfigs(Set<OriginalConfig> configs, int slave, String filenam
       }
    }
 
-   public void writeScenario(Collection<Report> reports) {
-      if (reports.isEmpty()) return;
-      // all reports should share the same stages
-      List<Report.Stage> stages = reports.iterator().next().getStages();
-      writeTag("h2", "Scenario");
-      write("Note that some properties may have not been resolved correctly as these depend on local properties<br>");
-      write("\nThese stages have been used for the benchmark:<br><ul>\n");
-      for (Report.Stage stage : stages) {
-         writeStage(stage);
-      }
-      write("</ul>\n");
+   private String escape(String str) {
+      return str.replaceAll("<", "&lt;").replace(">", "&gt;");
    }
 
-   private void writeStage(Report.Stage stage) {
-      writer.write("<li><span style=\"cursor: pointer\" onClick=\"");
-      List<Report.Property> properties = stage.getProperties();
-      for (int i = 0; i < properties.size(); ++i) {
-         writer.write("switch_li_display('e" + (elementCounter + i) + "');");
-      }
-      writer.write(String.format("\">%s</span><ul>\n", stage.getName()));
-      for (Report.Property property : properties) {
-         int propertyCounter = elementCounter++;
-         if (property.getDefinition() != null) {
-            writer.write(String.format("<li><strong>%s = %s</strong>&nbsp;" +
-                  "<small style=\"cursor: pointer\"  id=\"showdef%d\" onClick=\"switch_visibility('showdef%d'); switch_visibility('def%d');\">show definition</small>" +
-                  "<span id=\"def%d\" style=\"visibility: collapse\"><small style=\"cursor: pointer\" onClick=\"switch_visibility('showdef%d'); switch_visibility('def%d');\">hide definition:</small>" +
-                  "&nbsp;%s</span></li>\n", property.getName(), escape(String.valueOf(property.getValue())),
-                  propertyCounter, propertyCounter, propertyCounter, propertyCounter, propertyCounter, propertyCounter,
-                  escape(String.valueOf(property.getDefinition()))));
-         } else {
-            writer.write(String.format("<li id=\"e%d\" style=\"display: none\"><small>%s = %s</small></li>\n",
-                     propertyCounter, property.getName(), property.getValue()));
-         }
+   public Set<OriginalConfig> getConfigs(Report report) {
+      return configs.get(report);
+   }
+
+   public String expandableRows(int expandableRows) {
+      String result = "";
+      for (int i = 0; i < expandableRows; ++i) {
+         result += "switch_li_display('e" + (elementCounter + i) + "'); ";
```

Comment:
All HTML code should be placed in templates if possible, did you come across any limitations here?